test/e2e/framework/service_util.go

```diff
@@ -348,6 +348,22 @@ func PickNodeIP(c clientset.Interface) string {
 	return ip
 }
 
+// GetPodNodePairs return [Pod,Node] pairs for all pods in a namespace
+func GetPodNodePairs(c clientset.Interface, ns string) ([][2]string, error) {
```

Comment:
Nit: Call this PodNodePairs

Also I would recommend defining a struct for this and returning that rather than a [][]string

Reply:
IMO, given that this is an exported function, I would say its even more important to have a struct for this. It's going to be hard for a user of this function to understand how they should parse the return value. Using a struct makes it very explicit.